document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/vistors/PostgresDataAccessorIdentifierExpressionVisitor.java

```diff
@@ -0,0 +1,43 @@
+package org.hypertrace.core.documentstore.postgres.query.v1.vistors;
+
+import lombok.NoArgsConstructor;
+import org.hypertrace.core.documentstore.expression.impl.IdentifierExpression;
+import org.hypertrace.core.documentstore.postgres.utils.PostgresUtils;
+
+@NoArgsConstructor
+public class PostgresDataAccessorIdentifierExpressionVisitor
+    extends PostgresSelectTypeExpressionVisitor {
+
+  enum Type {
+    BOOL,
+    NUMERIC
+  }
+
+  private Type type = Type.NUMERIC;
+
+  public PostgresDataAccessorIdentifierExpressionVisitor(Type type) {
+    this.type = type;
+  }
+
+  public PostgresDataAccessorIdentifierExpressionVisitor(
+      PostgresSelectTypeExpressionVisitor baseVisitor) {
+    super(baseVisitor);
+  }
+
+  public PostgresDataAccessorIdentifierExpressionVisitor(
+      PostgresSelectTypeExpressionVisitor baseVisitor, Type type) {
+    super(baseVisitor);
+    this.type = type;
+  }
+
+  @Override
+  public String visit(final IdentifierExpression expression) {
+    String dataAccessor = PostgresUtils.prepareFieldDataAccessorExpr(expression.getName());
+    if (type.equals(Type.NUMERIC)) {
+      return PostgresUtils.prepareCast(dataAccessor, 1);
```

Comment:
Can we have separate `prepareCast()` methods that will take `Type` as argument instead of simply passing a dummy value of that type (and then inferring back that type)?

Reply:
Done